redis/src/types.rs

```diff
@@ -581,6 +581,32 @@ impl RedisError {
         };
         Self { repr }
     }
+
+    #[cfg(feature = "cluster")] // Used to avoid "unused method" warning
+    pub(crate) fn is_retryable(&self) -> bool {
```

Comment:
IMO this should be pub, not pub(crate) - this is usable to anyone who wants to implement a client over redis-rs. 
This is also a self contained change, that can be moved to a separate PR. Merging this early will make the review process for the larger PR faster, since it will reduce the amount of code to review.

Reply:
I think I'd like to leave this in the PR since integrating it into both clients will require a lot of the larger refactoring that's in here. I'll admit this is PR is a bit more unwieldy than I'd hoped and didn't quite end up where I intended it to :sweat_smile:. 

I'm not opposed to making `is_retryable` public, but for the moment I'm not sure whether all of the variants are handled properly so thought I'd avoid exposing it for now.